java/src/test/java/ai/rapids/cudf/DecimalColumnVectorTest.java

```diff
@@ -0,0 +1,187 @@
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package ai.rapids.cudf;
+
+import ai.rapids.cudf.HostColumnVector.Builder;
+import org.junit.jupiter.api.Test;
+
+import java.math.BigDecimal;
+import java.util.Random;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class DecimalColumnVectorTest extends CudfTestBase {
+  private final Random rdSeed = new Random();
+
+  private final BigDecimal[] decimal32Zoo = new BigDecimal[]{
+    BigDecimal.valueOf(rdSeed.nextInt(1000000000), 4),
+    BigDecimal.valueOf(rdSeed.nextInt(1000000000), 4),
+    BigDecimal.valueOf(rdSeed.nextInt(1000000000), 4),
+  };
+
+  private final BigDecimal[] decimal64Zoo = new BigDecimal[]{
+    BigDecimal.valueOf(rdSeed.nextLong(), 10),
+    BigDecimal.valueOf(rdSeed.nextLong(), 10),
+    null,
+    BigDecimal.valueOf(rdSeed.nextLong(), 10),
+  };
+
+  @Test
+  public void testCreateColumnVectorBuilder() {
+    try (ColumnVector decColumnVector = ColumnVector.build(DType.create(DType.DTypeEnum.DECIMAL32, -5), 3,
+        (b) -> b.append(BigDecimal.valueOf(123456789, 5)))) {
+      assertFalse(decColumnVector.hasNulls());
+    }
+    try (ColumnVector decColumnVector = ColumnVector.build(DType.create(DType.DTypeEnum.DECIMAL64, -10), 3,
+        (b) -> b.append(BigDecimal.valueOf(1023040506070809L, 10)))) {
+      assertFalse(decColumnVector.hasNulls());
+    }
+  }
+
+  @Test
+  public void testUpperIndexOutOfBoundsException() {
+    try (HostColumnVector decColumnVector = HostColumnVector.fromBigDecimals(decimal32Zoo)) {
+      assertThrows(AssertionError.class, () -> decColumnVector.getBigDecimal(3));
+      assertFalse(decColumnVector.hasNulls());
+    }
+  }
+
+  @Test
+  public void testLowerIndexOutOfBoundsException() {
+    try (HostColumnVector doubleColumnVector = HostColumnVector.fromBigDecimals(decimal32Zoo)) {
+      assertFalse(doubleColumnVector.hasNulls());
+      assertThrows(AssertionError.class, () -> doubleColumnVector.getBigDecimal(-1));
+    }
+  }
+
+  @Test
+  public void testAddingNullValues() {
+    try (HostColumnVector cv = HostColumnVector.fromBigDecimals(decimal64Zoo)) {
+      assertTrue(cv.hasNulls());
+      assertEquals(1, cv.getNullCount());
+      assertFalse(cv.isNull(0));
+      assertFalse(cv.isNull(1));
+      assertTrue(cv.isNull(2));
+      assertFalse(cv.isNull(3));
```

Comment:
Rather than hardcode this loop, I'd like to see it be more programmatically driven by the input, i.e.: check the corresponding value in`decimal64Zoo` to know whether to check for null or not.  Then we can simply change the element count in `decimal64Zoo` and this test keeps working as intended.

Reply:
Thanks for advice! Testcases are rewritten with a more programmatically way.